### DISCUSSION TITLE
could rpi pico w replace esp32 provifing both wireless and usb midi?

### DISCUSSION THREAD

**Discussion by zavorra** [alf45tar/PedalinoMini#446] [Ideas]
First of all, this project is incredible.

I recently purchased a mod dwarf and have read on their forum about it, as I need a midi over USB controller

I have noticed that an esp32 is used for the actual SW functionalities and for wifi/ble, and an Arduino/rp2040 for usb midi.

Could it be possible to use a single raspberry pi pico w to handle both midi processing, ble, wifi and USB without the need of an external MCU?

**Comment by alf45tar** [comment]
RP2040 has no BLE. I am trying to solve the external USB issue using a ESP32-S3 board. At the moment the software is not yet ready.
